src/libs/Network.js

```diff
@@ -25,25 +26,57 @@ let onError = () => {};
 let isOffline;
 Onyx.connect({
     key: ONYXKEYS.NETWORK,
-    callback: val => isOffline = val && val.isOffline,
-});
-
-let didLoadPersistedRequests;
-Onyx.connect({
-    key: ONYXKEYS.NETWORK_REQUEST_QUEUE,
-    callback: (persistedRequests) => {
-        if (didLoadPersistedRequests || !persistedRequests) {
+    callback: (val) => {
+        if (!val) {
             return;
         }
 
-        // Merge the persisted requests with the requests in memory then clear out the queue as we only need to load
-        // this once when the app initializes
-        networkRequestQueue = [...networkRequestQueue, ...persistedRequests];
-        didLoadPersistedRequests = true;
-        Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, []);
+        // Client becomes online, process the queue.
+        if (isOffline && !val.isOffline) {
+            // Fake merge which trigger the NETWORK_REQUEST_QUEUE Onyx's subscription callback
+            Onyx.merge(ONYXKEYS.NETWORK_REQUEST_QUEUE, []);
```

Comment:
@joelbettner This part triggers the Onyx.connect.

Reply:
So...somehow (I'm not clear on how) this triggers the `Onyx.connect` in `ActiveClientManager/index.js`?

Are you sure this isn't triggering the `Onyx.connect` that is within `ActiveClientManager.isReady().then` in `Network.js`?

Reply:
No. Onyx.connect in ActiveClientManager/index.js does not have any linking with being online. That tiggers Onyx.connect for NETWORK_REQUEST_QUEUE.

Reply:
Onyx.connect that is within ActiveClientManager.isReady(), will resolve when we add a new client id on the Active_clients key.

Reply:
I want to start a separate convo from the above discussion, and it's specifically about the use of a "fake merge" in order to trigger a callback. I'm concerned that this is a little bit of a "hack" and it's not clear what's really happening. Instead, I think it would be better like this:

1. Move the `ONYXKEYS.NETWORK_REQUEST_QUEUE` callback to be in its own method (not an anonymous function)
2. Reference the new function from `Onyx.connect()`
3. Call the new function at this spot.